packages/sdk/src/tests/getRequest.test.ts

```diff
@@ -0,0 +1,242 @@
+import "log-timestamp"
+
+import {
+	createPublicClient,
+	createWalletClient,
+	decodeFunctionData,
+	getContract,
+	hexToBytes,
+	http,
+	parseAbi,
+	parseEventLogs,
+} from "viem"
+import { privateKeyToAccount } from "viem/accounts"
+import { bscTestnet, gnosisChiado } from "viem/chains"
+
+import { IndexerClient } from "@/client"
+import { HexString, RequestStatus, TimeoutStatus } from "@/types"
+import { getRequestCommitment } from "@/utils"
+
+import ERC6160 from "@/abis/erc6160"
+import PING_MODULE from "@/abis/pingModule"
+import EVM_HOST from "@/abis/evmHost"
+import HANDLER from "@/abis/handler"
+import { EvmChain, SubstrateChain } from "@/chain"
+
+describe("GetRequest", () => {
+	let indexer: IndexerClient
+
+	beforeAll(async () => {
+		const { gnosisChiadoHost, bscIsmpHost } = await setUp()
+		indexer = new IndexerClient({
+			source: {
+				consensusStateId: "BSC0",
+				rpcUrl: process.env.BSC_CHAPEL!,
+				stateMachineId: "EVM-97",
+				host: bscIsmpHost.address,
+			},
+			dest: {
+				consensusStateId: "GNO0",
+				rpcUrl: process.env.GNOSIS_CHIADO!,
+				stateMachineId: "EVM-10200",
+				host: gnosisChiadoHost.address,
+			},
+			hyperbridge: {
+				consensusStateId: "PAS0",
+				stateMachineId: "KUSAMA-4009",
+				wsUrl: process.env.HYPERBRIDGE_GARGANTUA!,
+			},
+			url: "http://0.0.0.0:3100",
+			pollInterval: 1_000, // every second
+		})
+	})
+
+	it.only("should successfully stream and query the request status", async () => {
+		const { bscTestnetClient, gnosisChiadoHandler, bscPing, gnosisChiadoClient, gnosisChiadoHost, bscIsmpHost } =
+			await setUp()
+		console.log("\n\nSending Get Request\n\n")
+
+		const hash = await bscPing.write.dispatch([
+			{
+				source: await bscIsmpHost.read.host(),
+				dest: await gnosisChiadoHost.read.host(),
+				nonce: await bscIsmpHost.read.nonce(),
+				from: process.env.PING_MODULE_ADDRESS! as `0x${string}`,
+				timeoutTimestamp: BigInt(Math.floor(Date.now() / 1000) + 60 * 60),
+				keys: [
+					`0x${process.env.PING_MODULE_ADDRESS!.slice(2)}0000000000000000000000000000000000000000000000000000000000000001`,
+				],
+				height: await gnosisChiadoClient.getBlockNumber(),
```

Comment:
So this won't work. You'll need to query for a height thats already available on hyperbridge

Reply:
You can use the polkadotjs library for this

![Screenshot 2025-03-28 at 4 23 14 PM](https://github.com/user-attachments/assets/030e106c-aa6b-41db-b070-ea23e723876f)